app/src/main/java/org/p2p/wallet/swap/ui/jupiter/main/JupiterSwapFragment.kt

```diff
@@ -61,6 +64,14 @@ class JupiterSwapFragment :
             imageViewSwapTokens.setOnClickListener {
                 presenter.switchTokens()
             }
+            val onBackPressed = { presenter.onBackPressed() }
+            toolbar.setNavigationOnClickListener { onBackPressed() }
+            if (isInMainTabsScreen()) {
```

Comment:
i personally handle this cases with enum as an arg and checking here with if

it eliminates additional fragment logic and interfaces that you created

Reply:
imho easy to use extension than to create some extra_arg, extra_arg_key, add new method params, set param in other features or when parse deeplink